src/Microsoft.AspNet.TestHost/TestServer.cs

```diff
@@ -14,22 +14,26 @@
 
 namespace Microsoft.AspNet.TestHost
 {
-    public class TestServer : IServerFactory, IDisposable
+    public class TestServer : IServerFactory, IServer, IDisposable
     {
         private const string DefaultEnvironmentName = "Development";
         private const string ServerName = nameof(TestServer);
-        private static readonly IFeatureCollection ServerInfo = new FeatureCollection();
-        private Func<IFeatureCollection, Task> _appDelegate;
+        private RequestDelegate _appDelegate;
         private IDisposable _appInstance;
         private bool _disposed = false;
+        private IHttpContextAccessor _httpContextAccessor;
 
         public TestServer(WebHostBuilder builder)
         {
-            _appInstance = builder.UseServer(this).Build().Start();
+            var hostingEngine = builder.UseServer(this).Build();
+            _httpContextAccessor = hostingEngine.ApplicationServices.GetService<IHttpContextAccessor>();
+            _appInstance = hostingEngine.Start();
         }
 
         public Uri BaseAddress { get; set; } = new Uri("http://localhost/");
 
+        public IFeatureCollection Features { get; }
```

Comment:
Some of these interface members don't actually make sense here, use the explicit implementation to hide them:
`IServer.Features { get; }`